### PR TITLE
Fix #5810: category picker revamped

### LINKED ISSUE #5810: Post Settings: Redesign the Category Picker
The current version of the Category Picker is quite ugly. 

Step 1: use the material grid to align everything (the tree view and the "add new category" dialog). And fix the incoherent font sizes.
Step 2: make that tree view more androidy - update the indicator graphic, we might get some inspiration from https://material.io/guidelines/components/lists.html

### PR DESCRIPTION
**Pull request by maxme**
Fix #5810: category picker revamped

Updated the design to make it look like the iOS version with material specs (margins/text sizes).

Note: I removed the "category description" from the category creation dialog since this doesn't exist in Calypso and it feels superfluous.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
+package org.wordpress.android.ui.posts;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.content.Intent;
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java]: This import is unused.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import org.wordpress.android.fluxc.store.TaxonomyStore;
+import org.wordpress.android.models.CategoryNode;
+import org.wordpress.android.util.ToastUtils;
+
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
+public class AddCategoryFragment extends AppCompatDialogFragment {
+    public static final String KEY_CATEGORY = "KEY_CATEGORY";
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java]: This constant is unused.

**Comment by nbradbury** [comment]
The app crashes if you rotate the device while the "add category" dialog is showing.

```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.app.Dialog.cancel()' on a null object reference
                                                                              at org.wordpress.android.ui.posts.AddCategoryFragment.initSite(AddCategoryFragment.java:89)
                                                                              at org.wordpress.android.ui.posts.AddCategoryFragment.onCreateDialog(AddCategoryFragment.java:45)
                                                                              at android.support.v4.app.DialogFragment.getLayoutInflater(DialogFragment.java:312)
                                                                              at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299)
                                                                              at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
                                                                              at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
                                                                              at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2900)
                                                                              at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:201)
                                                                              at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:603)
                                                                              at android.support.v7.app.AppCompatActivity.onStart(AppCompatActivity.java:178)
```

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 3]
```diff
@@ -216,12 +215,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
         mExcerptTextView = (TextView) rootView.findViewById(R.id.post_excerpt);
         mSlugTextView = (TextView) rootView.findViewById(R.id.post_slug);
+        mCategoriesTextView = ((TextView) rootView.findViewById(R.id.post_categories));
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: Minor nit: there's an unnecessary set of parens here.

**Comment by nbradbury** [comment]
This dialog appears if you click OK in the "add category" dialog without entering anything.

screenshot_1497025150

I think we can get rid of this and simply not close the dialog in this situation.

**Comment by maxme** [comment]
- [x] remove unused code.
- [x] fix crash on rotate.
- [x] display "empty category" error in the Text view instead of showing a toast.

@nbradbury ready for another round

**Comment by nbradbury** [comment]
Not sure if you want to address this now but if I attempt to add a category that already exists, the dialog closes and then I see a toast saying "Adding category failed." It would be nice if the dialog could detect the duplication and show a TextView error instead.

**Comment by maxme** [comment]
> Not sure if you want to address this now but if I attempt to add a category that already exists, the dialog closes and then I see a toast saying "Adding category failed." It would be nice if the dialog could detect the duplication and show a TextView error instead.

That would mean, we have to make the dialog wait for the network response, and show some sort of progress bar / animation. I don't want to change the logic that much for a very minor improvement on error handling.

**Comment by nbradbury** [comment]
> That would mean, we have to make the dialog wait for the network response, and show some sort of progress bar / animation.

Well, I was thinking it could just check if the category exists in the local cache, but you're right that it's a minor improvement. I'm good to merge without that change. :shipit:

**Pull request merged**